feat: 계획 블록 삭제

m1njae opened the pull request:
<!-- 
- PR 제목: [feat/fix/refactor...] 작업 내용 한 줄 요약 (브랜치 이름)
- commit message가 적절한지 확인해주세요.
- 적절한 branch로 요청했는지 확인해주세요.
- Assignees, Label을 붙여주세요.
- 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer로 등록해주세요.
- PR이 승인된 경우 해당 브랜치는 삭제 부탁드립니다. 
-->


## 📝 PR 요약

<!--
해당 pr에서 작업한 내역을 적어주세요.
-->

- 일간, 우회할, 루틴로드 계획을 삭제합니다.

<br/>

## 📌 변경 사항 및 주의 사항

<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->
- 계획 블록을 삭제 하는 과정에서 생각해볼 에러상황들이 있는 것 같아서 일단 모두 다 고려해보았습니다!
- 로직의 흐름은 주석을 달아놓았으니 확인해보시면 될 거 같습니다
- 에러 처리를 한다고 했는데 뭔가 아쉬워보이긴 하네요.. 혹시 더 좋은 에러핸들링이 있다면 알려주세욥!
- 점점 git이 꼬여간다고 느끼는 것은 저의 기분 탓이겠지만 체다나와 크라켄만 있다면 모두 해결할 수 있을 거 같습니다 하핳
- 피드백은 언제나 환영입니다:)

![image](https://user-images.githubusercontent.com/82744423/213910642-de2df005-3271-41ca-83ce-2493afa15f74.png)



<br/>

## 🔗 Linked Issue
close #17

m1njae committed:
add: responseMessage 추가

m1njae committed:
feat: 계획 블록 삭제 구현

Chedda98 commented on src/services/PlanService.ts:
```diff
@@ -44,4 +44,57 @@ export class DailyPlanService {
+          planDate: date,
+        },
+        {
+          planList: updatePlanList,
```
@m1njae 민재 이건 개인적으로 몰라서 여쭤보고 싶어 코멘트 남깁니다!!
**planId가 제거된 배열을 DB에 저장하는 이유**가 따로 있을까요?!!

m1njae replied:
계획블록을 온전하게 삭제하기 위해서 해야할 일은 두 가지라고 생각했습니다!
1. 순서 이동과 변경을 관리해주는 planOrder 테이블에서 planList 안에 planId를 지워준다.
2. plan 테이블에서 계획 블록을 삭제한다.

planId가 제거된 배열을 DB에 업데이트 해야한다고 생각했던 이유는 삭제하고자 하는 계획블록이 planList에서 없어지더라도 이후에 planList에 있는 다른 계획블록들의 순서가 지장이 없어야 하기 때문입니다!
-> 그래서 totalPlan을 통해서 삭제해야 할 id가 있는 planList를 찾아내고 그 planList에서 planId를 제거하는 과정을 거친 후, 그 배열을 다시 DB에 업데이트 해주는 형식으로 구현했습니당

제가 말을 조금 장황하게 풀어낸거 같은데 이해가 되었을까요..?

m1njae committed:
chore: merge 이후 코드 정리

m1njae committed:
add: deletePlanValidation 추가

m1njae committed:
chore: planController 코드 정리

m1njae committed:
chore: planService 로직 수정

daily 시에는 planDate 값이 필요로 하지만, 시간 데이터를 저장하지 않고 있는 루틴로드와 우회할 계획의 경우,
시간 데이터가 필요하지 않습니다. 따라서 분기처리하여 로직을 수정하였습니다.
